File: kloster/model/beercollection.go
package model

import (
	"bitbucket.org/ctiu6442/m14b-group-1-cp55/kloster/id"
	"time"
)

type BeerCollectionItem struct {
	BeerID              id.BeerID `json:"beerID"`
	StorageLocation     string    `json:"storageLocation"`
	PurchasePrice       float64   `json:"purchasePrice"`
	PurchaseDate        time.Time `json:"purchaseDate"`
	PurchaseDistributor string    `json:"purchaseDistributor"`
	BottlesNumber       int       `json:"bottlesNumber"`
	BestBeforeDate			time.Time	`json:"bestBeforeDate"`
}

type BeerCollection struct {
	Name        string               `json:"name"`
	Description string               `json:"description"`
	Tags        []string             `json:"tags"`
	Location    string               `json:"location"`
	Notes       string               `json:"notes"`
	Items       []BeerCollectionItem `json:"items"`
}

File: paleale/src/model/gomapper/beercollection.ts
import {List} from "immutable";
import {DateTime} from "luxon";

import {BeerCollection, BeerCollectionItem} from "../beercollection";

export function toBeerCollectionItem(item: any): BeerCollectionItem {
  return {
    beerID: item.beerID,
    bestBeforeDate: DateTime.fromISO(item.bestBeforeDate),
    bottlesNumber: item.bottlesNumber,
    purchaseDate: DateTime.fromISO(item.purchaseDate),
    purchaseDistributor: item.purchaseDistributor,
    purchasePrice: item.purchasePrice,
    storageLocation: item.storageLocation,
  };
}

export function toBeerCollection(collection: any): BeerCollection {
  return {
    description: collection.description,
    items: List(collection.items).map(toBeerCollectionItem),
    location: collection.location,
    name: collection.name,
    notes: collection.notes,
    tags: List<string>(collection.tags),
  };
}
